src/common/CalculatorEngineCommon/ExprtkEvaluator.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ExprtkEvaluator.h"
+#include "exprtk.hpp"
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace ExprtkCalculator::internal
+{
+
+    std::wstring ToWStringFullPrecision(double value)
+    {
+        std::wostringstream oss;
+        oss << std::fixed << std::setprecision(15) << value;
+        return oss.str();
+    }
+
+    std::string WStringToAscii(const std::wstring& wstr)
+    {
+        std::string result;
+        result.reserve(wstr.size());
+        for (wchar_t wc : wstr)
+            result.push_back(static_cast<char>(wc));
+        return result;
+    }
+
+    std::wstring EvaluateExpression(
+        const std::wstring& expressionText,
+        const std::unordered_map<std::wstring, double>& constants)
+    {
+        exprtk::symbol_table<double> symbol_table;
+
+        for (auto const& [name, value] : constants)
+        {
+            symbol_table.add_constant(WStringToAscii(name.c_str()), value);
+        }
+
+        exprtk::expression<double> expression;
+        expression.register_symbol_table(symbol_table);
+
+        exprtk::parser<double> parser;
```

Comment:
@lei9444 :  I recommend disabling the features in ExprTk that you think wont be required for the types of expressions that are expected to be evaluated. As an example will _loops_ and _conditional statements_ be required?

Disabling features can be done easily via the parser instance. **Section 19** of the documentation has all the necessary details:

https://www.partow.net/programming/exprtk/readme.html#line_3595

Reply:
Thanks a lot for your quick review and helpful feedback! Really appreciate your support on this.